pilot/pkg/xds/ads_common.go

```diff
@@ -23,20 +23,19 @@ import (
 // configKindAffectedProxyTypes contains known config types which will affect certain node types.
 var configKindAffectedProxyTypes = map[config.GroupVersionKind][]model.NodeType{
 	gvk.Gateway: {model.Router},
-	gvk.Secret:  {model.Router},
+	//gvk.Secret:  {model.Router},
```

Comment:
This is one of the last TODOs I had before marking this PR not WIP. This should be reverted

Reply:
Most of the pilot/pkg/xds changes are broken out to https://github.com/istio/istio/pull/29794, so we won't need to review them here if that is merged there. Its a lot of boilerplate since a common interface needed to change